server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -433,16 +429,16 @@ public TSExecuteBatchStatementResp executeBatchStatement(TSExecuteBatchStatement
       }
 
       if (isAllSuccessful) {
-        return getTSBathExecuteStatementResp(TS_StatusCode.SUCCESS_STATUS,
+        return getTSBatchExecuteStatementResp(TS_StatusCode.SUCCESS_STATUS, SUCCESS_CODE,
             "Execute batch statements successfully", result);
       } else {
-        return getTSBathExecuteStatementResp(TS_StatusCode.ERROR_STATUS,
+        return getTSBatchExecuteStatementResp(TS_StatusCode.ERROR_STATUS, ErrorCode.EXECUTE_STATEMENT_ERROR,
```

Comment:
It feels a bit strange to have StatusCode and ErrorCode at the same time, maybe you can synthesize them.

Reply:
Hi,

The existing code defined `statusCode` and `errorCode` separately, and `errorCode` is only set when `statusCode` is `ERROR_STATUS`:
```
struct TS_Status {
  1: required TS_StatusCode statusCode
  ...
  // If status is ERROR, then the following fields may be set
  4: optional i32 errorCode // internal error code
  5: optional string errorMessage
}
```
Should I change this struct by combine statusCode and errorCode (and maybe also include errorMessage) ?

Reply:
Yes, those fields seem a bit redundant now.